contents/docs/product-analytics/sql.md

```diff
@@ -0,0 +1,293 @@
+- `toDayOfWeek`, `toHour`, `toMinute`: Converts date number of day of week (1-7), hour in 24-hour time (0-23), and minute in hour (0-59).
+- `toStartOfYear`, `toStartOfMonth`, `toMonday`, `toStartOfDay`, `toStartOfMinute`: rounds date down to the nearest year, month, Monday, day, hour, or minute respectively
+- `dateDiff('unit', startdate, enddate)`: Returns the count in `unit` between `startdate` and `enddate`.
+- `formatDateTime`: Formats a time according to a [MySQL datetime format string](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-format).
+
+```sql
+SELECT 
+   formatDateTime(now(), '%a %b %T') AS current_time,
+   toDayOfWeek(now()) AS current_day_of_week,
+   dateDiff('day', timestamp, now()) AS days_since_event
+FROM events
+WHERE timestamp > now() - interval 1 day
+```
+
+> Read more examples in [How to do time-based breakdowns (hour, minute, real time)](/tutorials/time-breakdowns) and [Using HogQL for advanced time and date filters](/tutorials/hogql-date-time-filters).
+
+### String
+
+- `extract`: Extracts a fragment of a string using a regular expression.
+- `concat`: Concatenates strings listed without separator.
+- `splitByChar`, `splitByString`, `splitByRegexp`, `splitByWhitespace`: splits a string into substring separated by a specified character, string, regular expression, or whitespace character respectively.
+- `match`: Return whether the string matches a regular expression pattern.
+- `replaceOne`, `replaceRegexpOne`: Replace the first occurrence of matching a substring or regular expression pattern respectively with a replacement string.
+- `trim`: Remove specified characters (or whitespace) from the start or end of a string.
+
+```sql
+select 
+   extract(elements_chain, '[:|"]attr__class="(.*?)"') as class_name,
+	 concat(properties.$os, ' version: ', properties.$os_version),
+	 replaceRegexpOne(properties.$current_url, '^/', 'site/') AS modified_current_url
+from events
+where event = '$autocapture'
+```
+
+### Sparkline
+
+A sparkline is a tiny graph contained in one cell of your query result. As an argument, it takes an array of integers.
+
+```sql
+SELECT sparkline(range(1, 10)) FROM (SELECT 1)
+```
+
+You can use it to visualize queries, such as a 24-hour `$pageview` count for different `$current_url` values.
+
+```sql
+SELECT
+    pageview,
+    sparkline(arrayMap(h -> countEqual(groupArray(hour), h), range(0,23))),
+    count() as pageview_count
+FROM
+(
+    SELECT
+        properties.$current_url as pageview,
+        toHour(timestamp) AS hour
+    FROM
+        events
+    WHERE
+         timestamp > now () - interval 1 day
+         and event = '$pageview'
+) subquery
+GROUP BY
+    pageview
+ORDER BY
+    pageview_count desc
+```
+
+You can also use it for art.
+
+```sql
+select 
+    sparkline(arrayMap(a -> cos(toSecond(timestamp) + a/4), range(100 + 5 * toSecond(timestamp)))) 
+from events
+```
```

Comment:
Great docs. Optional: I think it would be great to include a section on "Useful examples", where you could put these docs in practice and show some of the common SQL insights people would create

Reply:
For this one, I think I will leave as is for now as:

1. I've already added some basic examples.
2. Tutorials cover examples pretty well.

Could add more later if it's still a problem